examples/07_extra-hono-router/src/components/Form.tsx

```diff
@@ -0,0 +1,127 @@
+'use client';
+
+import { useActionState, useCallback, useEffect, useState } from 'react';
+import { useFormStatus } from 'react-dom';
+import { client } from '../rpc-client';
+
+const SubmitButton = () => {
+  const { pending } = useFormStatus();
+
+  return (
+    <button
+      type="submit"
+      className="hover:bg-slate-50 w-fit rounded-lg border border-gray-200 bg-white p-2 transition-colors duration-200 disabled:opacity-50"
+      disabled={pending}
+    >
+      {pending ? 'Saving...' : 'Save Name'}
+    </button>
+  );
+};
+
+const GetNameButton = ({
+  isLoading,
+  onClick,
+}: {
+  isLoading: boolean;
+  onClick: () => void;
+}) => {
+  return (
+    <button
+      type="button"
+      onClick={onClick}
+      className="hover:bg-slate-50 w-fit rounded-lg border border-gray-200 bg-white p-2 transition-colors duration-200 disabled:opacity-50"
+      disabled={isLoading}
+    >
+      {isLoading ? 'Loading...' : 'Get Name'}
+    </button>
+  );
+};
+
+export const NameForm = () => {
+  const [isGetting, setIsGetting] = useState(false);
+  const [name, setName] = useState<string>();
+
+  useEffect(() => {
+    setTimeout(() => {
+      setName(undefined);
+    }, 1500);
+  }, [name]);
+
+  const formAction = async (prevState: any, formData: FormData) => {
+    try {
+      const name = formData.get('name');
+      const response = await client.api.hono.hello.$post({
+        json: { name },
+      });
+      const data = await response.json();
+      setName(data.message);
+      return { message: data.message, error: null };
+    } catch {
+      setName('Failed to save name');
+      return { message: null, error: 'Failed to save name' };
+    }
+  };
+
+  const [_, action] = useActionState(formAction, {
+    message: 'form action',
+    error: null,
+  });
+
+  const handleGetName = useCallback(async () => {
+    setIsGetting(true);
+    try {
+      const response = await client.api.hono.hello.$get();
+      const data = await response.json();
+      setName(data.message);
+    } catch {
+      setName('Failed to get name');
+    } finally {
+      setIsGetting(false);
+    }
+  }, []);
```

Comment:
I should probably refactor this to use tanstack query if we're ok with adding the library to this example